### PR TITLE
Spotaut 18517 - Implemented Go Sdk for Verification Template

### PR DESCRIPTION
**Pull request by IGUDE2**
_This is the default pull request template. You can customize it by adding a `pull_request_template.md` at the root of your repo or inside the `.github` folder._

# Jira Ticket

_Include a link to your Jira Ticket_
_Example: [JIRAISS-1234](https://spotinst.atlassian.net/browse/JIRAISS-1234)_

# Demo

_Please add a recording of the feature/bug fix in work. if you added new routes, the recording should show the request and response for each new/changed route_

# Checklist:
- [ ] I have filled relevant self assessment ([NodeJS](https://docs.google.com/forms/d/e/1FAIpQLSfl14u9AOBAmxVJ272tvO7XNuXE-EMvWaGcaRZalu1UAKB7RA/viewform), [Frontend](https://docs.google.com/forms/d/e/1FAIpQLSdiBPNKH81w_EkavihVL8Uwb0j7tP8PwJmLFYm2nCOQxz-1qw/viewform), [Backend](https://docs.google.com/forms/d/e/1FAIpQLSed_PsTJ5-XIWkFL6BSDE2AQRBVPmwc3PAmHZkUn-erzVI37Q/viewform?usp=sf_link))
- [ ] I have run ESlint on my changes and fixed all warnings and errors (**NodeJS & Frontend Services**)
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have validated all the requirements in the Jira task were answered
- [ ] I have all neccessary approvals for the design/mini design of this task
- [ ] I have approved the API changes and granular permission patterns (documentation subtask) (**For public services only**)

### CODE REVIEW

File: service/oceancd/verificationTemplate.go [hunk 1]
```diff
@@ -0,0 +1,1100 @@
...
+	nullFields      []string
+}
+
+type Provider struct {
+	CloudWatch *CloudWatchProvider `json:"cloudWatch,omitempty"`
+	DataDog    *DataDogProvider    `json:"dataDog,omitempty"`
+	Jenkins    *JenkinsProvider    `json:"jenkins,omitempty"`
+	NewRelic   *NewRelicProvider   `json:"newRelic,omitempty"`
+	Prometheus *PrometheusProvider `json:"prometheus,omitempty"`
+	Web        *Web                `json:"web,omitempty"`
```
Comment (chandra1-n) [inline service/oceancd/verificationTemplate.go:86]: job object is missing

File: service/oceancd/verificationTemplate.go [hunk 2]
```diff
@@ -0,0 +1,1100 @@
...
+
+	forceSendFields []string
+	nullFields      []string
+}
+
+type Metrics struct {
+	Baseline              *Baseline `json:"baseline,omitempty"`
+	ConsecutiveErrorLimit *int      `json:"consecutiveErrorLimit,omitempty"`
+	Count                 *int      `json:"count,omitempty"`
+	DryRun                *bool     `json:"dryRUn,omitempty"`
```
Comment (chandra1-n) [inline service/oceancd/verificationTemplate.go:57]: correct the dryRUn to dryRun

File: service/oceancd/verificationTemplate.go [hunk 3]
```diff
@@ -0,0 +1,1100 @@
...
+	Provider  *Provider `json:"provider,omitempty"`
+	Threshold *string   `json:"threshold,omitempty"`
+
+	forceSendFields []string
+	nullFields      []string
+}
+
+type Provider struct {
+	CloudWatch *CloudWatchProvider `json:"cloudWatch,omitempty"`
+	DataDog    *DataDogProvider    `json:"dataDog,omitempty"`
```
Comment (chandra1-n) [inline service/oceancd/verificationTemplate.go:82]: corect dataDog to datadog

File: service/oceancd/verificationTemplate.go [hunk 4]
```diff
@@ -0,0 +1,1100 @@
...
+
+	forceSendFields []string
+	nullFields      []string
+}
+
+type MetricDataQueries struct {
+	Expression *string     `json:"expression,omitempty"`
+	ID         *string     `json:"id,omitempty"`
+	Label      *string     `json:"label,omitempty"`
+	MetricStat *MetricStat `json:"MetricStat,omitempty"`
```
Comment (chandra1-n) [inline service/oceancd/verificationTemplate.go:104]: modify to metricStat

File: service/oceancd/verificationTemplate.go [hunk 5]
```diff
@@ -0,0 +1,1100 @@
...
+	Period     *int        `json:"duration,omitempty"`
```
Comment (chandra1-n) [inline service/oceancd/verificationTemplate.go:105]: modify to period

File: service/oceancd/verificationTemplate.go [hunk 6]
```diff
@@ -0,0 +1,1100 @@
...
+	ReturnData *bool       `json:"returnData,omitempty"`
+
+	forceSendFields []string
+	nullFields      []string
+}
+
+type MetricStat struct {
+	Metric *Metric `json:"address,omitempty"`
```
Comment (chandra1-n) [inline service/oceancd/verificationTemplate.go:113]: modify to metric

File: service/oceancd/verificationTemplate.go [hunk 7]
```diff
@@ -0,0 +1,1100 @@
...
+	Unit   *string `json:"unit,omitempty"`
+
+	forceSendFields []string
+	nullFields      []string
+}
+
+type Metric struct {
+	Dimensions []*Dimensions `json:"dimensions,omitempty"`
+	MetricName *string       `json:"metricName,omitempty"`
+	NameSpace  *string       `json:"nameSpace,omitempty"`
```
Comment (chandra1-n) [inline service/oceancd/verificationTemplate.go:125]: shouldn't be camel-case

File: service/oceancd/verificationTemplate.go [hunk 8]
```diff
@@ -0,0 +1,1100 @@
...
+type DataDogProvider struct {
+	Duration *string `json:"duration,omitempty"`
+	Query    *string `json:"query,omitempty"`
+
+	forceSendFields []string
+	nullFields      []string
+}
+
+type JenkinsProvider struct {
+	Interval        *string       `json:"address,omitempty"`
```
Comment (chandra1-n) [inline service/oceancd/verificationTemplate.go:148]: modify to interval

File: service/oceancd/verificationTemplate.go [hunk 9]
```diff
@@ -0,0 +1,1100 @@
...
+	nullFields      []string
+}
+
+type Web struct {
+	Body           *string    `json:"body,omitempty"`
+	Insecure       *bool      `json:"insecure,omitempty"`
+	Headers        []*Headers `json:"headers,omitempty"`
+	JsonPath       *string    `json:"jsonPath,omitempty"`
+	Method         *string    `json:"method,omitempty"`
+	TimeoutSeconds *int       `json:"TimeoutSeconds,omitempty"`
```
Comment (chandra1-n) [inline service/oceancd/verificationTemplate.go:187]: modify to timeoutSeconds

**Review by chandra1-n — CHANGES_REQUESTED** [review]
please fix the review comments

File: service/oceancd/verificationTemplate.go (hunk 1 above)
Comment (IGUDE2) [reply service/oceancd/verificationTemplate.go:86]: Added Job Object and definations

File: service/oceancd/verificationTemplate.go [hunk 10]
```diff
@@ -0,0 +1,1100 @@
...
+	MaxRange  *int      `json:"maxRange,omitempty"`
+	MinRange  *int      `json:"minRange,omitempty"`
+	Provider  *Provider `json:"provider,omitempty"`
+	Threshold *string   `json:"threshold,omitempty"`
+
+	forceSendFields []string
+	nullFields      []string
+}
+
+type Provider struct {
```
[the remark this answers is not in the document]
Comment (IGUDE2) [reply service/oceancd/verificationTemplate.go:80]: Restested with the Provider

File: service/oceancd/verificationTemplate.go (hunk 2 above)
Comment (IGUDE2) [reply service/oceancd/verificationTemplate.go:57]: corrected

File: service/oceancd/verificationTemplate.go (hunk 3 above)
Comment (IGUDE2) [reply service/oceancd/verificationTemplate.go:82]: corrected

File: service/oceancd/verificationTemplate.go (hunk 4 above)
Comment (IGUDE2) [reply service/oceancd/verificationTemplate.go:104]: corrected

File: service/oceancd/verificationTemplate.go (hunk 5 above)
Comment (IGUDE2) [reply service/oceancd/verificationTemplate.go:105]: modified

File: service/oceancd/verificationTemplate.go (hunk 6 above)
Comment (IGUDE2) [reply service/oceancd/verificationTemplate.go:113]: modified

File: service/oceancd/verificationTemplate.go (hunk 7 above)
Comment (IGUDE2) [reply service/oceancd/verificationTemplate.go:125]: changed

File: service/oceancd/verificationTemplate.go (hunk 8 above)
Comment (IGUDE2) [reply service/oceancd/verificationTemplate.go:148]: modified

File: service/oceancd/verificationTemplate.go (hunk 9 above)
Comment (IGUDE2) [reply service/oceancd/verificationTemplate.go:187]: modified

**Review by chandra1-n — APPROVED** [review]

**Pull request merged**